### PR TITLE
AP_Soaring: Vario filter cleanup

### PR DESCRIPTION
**Pull request by samuelctabor**
- Convert in-line filters to LowPassFilter instances
- Rename variables and cleanup access

### CODE REVIEW

**Comment by samuelctabor** [comment]
Any comments on this @tridge ? I know you made some fixes to the use of LowPassFilter in AP_Soaring recently.

**Comment by samuelctabor** [comment]
Dev call EU?

File: libraries/AP_Soaring/Variometer.cpp [hunk 1]
```diff
@@ -22,12 +22,12 @@ void Variometer::update(const float thermal_bank, const float polar_K, const flo
...
     }
-    _aspd_filt = _sp_filter.apply(aspd);
+
+    float aspd_filt = _sp_filter.apply(aspd);
 
     // Constrained airspeed.
     const float minV = sqrtf(polar_K/1.5);
-    _aspd_filt_constrained = _aspd_filt>minV ? _aspd_filt : minV;
-
+    _aspd_filt_constrained = aspd_filt>minV ? aspd_filt : minV;
```
Comment (peterbarker) [inline libraries/AP_Soaring/Variometer.cpp:30]: We do supply `MIN` for this purpose.

File: libraries/AP_Soaring/Variometer.cpp [hunk 2]
```diff
@@ -43,20 +43,22 @@ void Variometer::update(const float thermal_bank, const float polar_K, const flo
...
+
+    float raw_climb_rate = 0.0f;
     if (_ahrs.get_velocity_NED(velned)) {
         // if possible use the EKF vertical velocity
         raw_climb_rate = -velned.z;
     }
     
     _climb_filter.set_cutoff_frequency(1/(3*tau));
-    smoothed_climb_rate = _climb_filter.apply(raw_climb_rate, dt);
+    float smoothed_climb_rate = _climb_filter.apply(raw_climb_rate, dt);
```
Comment (peterbarker) [inline libraries/AP_Soaring/Variometer.cpp:61]:
```suggestion
    const float smoothed_climb_rate = _climb_filter.apply(raw_climb_rate, dt);
```

File: libraries/AP_Soaring/Variometer.cpp [hunk 3]
```diff
@@ -65,8 +67,9 @@ void Variometer::update(const float thermal_bank, const float polar_K, const flo
     reading = raw_climb_rate + dsp_cor*_aspd_filt_constrained/GRAVITY_MSS + sinkrate;
     
 
-    filtered_reading = TE_FILT * reading + (1 - TE_FILT) * filtered_reading;                       // Apply low pass timeconst filter for noise
-    displayed_reading = TE_FILT_DISPLAYED * reading + (1 - TE_FILT_DISPLAYED) * displayed_reading;
+    float filtered_reading = _trigger_filter.apply(reading, dt); // Apply low pass timeconst filter for noise
```
Comment (peterbarker) [inline libraries/AP_Soaring/Variometer.cpp:70]:
```suggestion
    const float filtered_reading = _trigger_filter.apply(reading, dt); // Apply low pass timeconst filter for noise
```

**Review by peterbarker — APPROVED** [review]
LGTM

**Comment by peterbarker** [comment]
Merged, thanks!

**Comment by samuelctabor** [comment]
Thanks Peter!

**Pull request merged**